Optimize order_swich_ops on GPU

xiaomengy opened the pull request:
Summary: Optimize order_swich_ops on GPU

Differential Revision: D9728642

BIT-silence commented:
@pytorchbot retest this please

BIT-silence commented:
@pytorchbot retest this please

houseroad commented on caffe2/operators/order_switch_ops_cudnn.cc:
```diff
@@ -0,0 +1,160 @@
+      const int W = order == StorageOrder::NCHW ? data_dims[3] : data_dims[2];
+      CUDNN_ENFORCE(cudnnSetTensor4dDescriptor(
+          data_desc, GetCudnnTensorFormat(order), data_type, N, C, H, W));
+    } else {
```
shall we turn this to else if (ndim == 5) and add else to throw the error if it's not supported by CuDNN?

BIT-silence replied:
It is fine to combine the dims greater than 5 together. It may not be as fast as 5d, but there will be almost no such cases. So I think it's fine to keep it as that.

houseroad reviewed (approved):
Overall LG!

BIT-silence commented:
@pytorchbot retest this please

xiaomengy committed:
Optimize order_swich_ops on GPU (#11404)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/11404

Optimize order_swich_ops on GPU

Reviewed By: houseroad

Differential Revision: D9728642

fbshipit-source-id: e5b5ea296146a045c6d8a9c74078d74236e2262b
